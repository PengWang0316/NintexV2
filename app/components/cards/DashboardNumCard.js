import React from 'react';
import PropTypes from 'prop-types';
import { Box, Card, CardContent } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  card: {
    width: 255,
    margin: '0 10px',
  },
  cardDiv: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 20,
  },
  contentDiv: {
    display: 'flex',
    justifyContent: 'space-between',
  },
};
// The regular experssion to help format numbers to have a comma in every three digitals
const NUM_FORMATE_REGEXP = /(\d)(?=(\d{3})+(?!\d))/g;

export const DashboardNumCard = ({
  classes, displayNum, title, icon, extraContent,
}) => (
  <Card className={classes.card}>
    <CardContent>
      <div className={classes.contentDiv}>
        <div>
          <Box fontSize={14} color="textSecondary">
            {title}
          </Box>
          <Box fontSize={26} fontWeight={1600} color="textSecondary">
            {displayNum ? displayNum.toString().replace(NUM_FORMATE_REGEXP, '$1,') : '--'}
          </Box>
        </div>
        {icon}
      </div>
      {extraContent}
    </CardContent>
  </Card>
);
DashboardNumCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  displayNum: PropTypes.number,
  title: PropTypes.string.isRequired,
  icon: PropTypes.objectOf(PropTypes.any).isRequired,
  extraContent: PropTypes.element,
};
DashboardNumCard.defaultProps = { displayNum: null, extraContent: null };
export default withStyles(styles)(DashboardNumCard);
