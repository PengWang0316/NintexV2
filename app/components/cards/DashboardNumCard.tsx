import React, { ReactElement, memo } from 'react';
import { Box, Card, CardContent } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

interface Props {
  displayNum: string | number | null;
  title: string;
  icon: ReactElement;
  extraContent?: ReactElement | null;
}

const useStyles = makeStyles({
  card: {
    width: 255,
    margin: '0 10px',
  },
  cardDiv: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 20,
  },
  contentDiv: {
    display: 'flex',
    justifyContent: 'space-between',
  },
});
// The regular experssion to help format numbers to have a comma in every three digitals
const NUM_FORMATE_REGEXP = /(\d)(?=(\d{3})+(?!\d))/g;

export const DashboardNumCard = ({
  displayNum = null, title, icon, extraContent = null,
}: Props): ReactElement => {
  const classes = useStyles({});
  return (
    <Card className={classes.card}>
      <CardContent>
        <div className={classes.contentDiv}>
          <div>
            <Box fontSize={14} color="textSecondary">
              {title}
            </Box>
            <Box fontSize={26} fontWeight={1600} color="textSecondary">
              {displayNum ? displayNum.toString().replace(NUM_FORMATE_REGEXP, '$1,') : '--'}
            </Box>
          </div>
          {icon}
        </div>
        {extraContent}
      </CardContent>
    </Card>
  );
};
export default memo(DashboardNumCard);
