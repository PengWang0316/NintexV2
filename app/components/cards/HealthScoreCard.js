/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightBlue, red, lightGreen } from '@material-ui/core/colors';
import { SwapVerticalCircle } from '@material-ui/icons';
import { Box } from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceStatus as fetchInstanceStatusAction } from '../../actions/InstanceActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: lightBlue[500],
  },
  flexDiv: { display: 'flex' },
  box: { marginRight: 10 },
};

export const HealthScoreCard = ({ classes, instanceStatus, fetchInstanceStatus }) => {
  useEffect(() => {
    if (!instanceStatus) fetchInstanceStatus();
  });

  return (
    <DashboardNumCard
      displayNum={instanceStatus ? instanceStatus.completedPercentage : null}
      title={I18n.get('cardTitleWorkflowHealth')}
      icon={<SwapVerticalCircle className={classes.cardIcon} />}
      extraContent={(
        <div className={classes.flexDiv}>
          <Box className={classes.box} fontSize={14} color={red[500]}>Failed: {instanceStatus ? instanceStatus.failedPercentage : '--'}</Box>
          <Box fontSize={14} color={lightGreen[500]}>Started: {instanceStatus ? instanceStatus.startedPercentage : '--'}</Box>
        </div>
      )}
    />
  );
};

HealthScoreCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  instanceStatus: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])),
  fetchInstanceStatus: PropTypes.func.isRequired,
};
HealthScoreCard.defaultProps = {
  instanceStatus: null,
};
const mapStateToProps = state => ({ instanceStatus: state.instanceStatus });
const mapDispatchToProps = { fetchInstanceStatus: fetchInstanceStatusAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(HealthScoreCard));
