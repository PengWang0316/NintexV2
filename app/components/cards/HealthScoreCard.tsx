/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect, memo } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightBlue, red, lightGreen } from '@material-ui/core/colors';
import { SwapVerticalCircle } from '@material-ui/icons';
import { Box } from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceStatus as fetchInstanceStatusAction } from '../../store/InstanceStatus/actions';
import { AppState } from '../../store/ConfigureStore';

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: lightBlue[500],
  },
  flexDiv: { display: 'flex' },
  box: { marginRight: 10 },
});
let isFetching = false;

interface Props {
  instanceStatus: {
    completedPercentage: string | number | null;
    failedPercentage: string | number;
    startedPercentage: string | number;
  };
  fetchInstanceStatus: () => void;
}

export const HealthScoreCard = ({ instanceStatus, fetchInstanceStatus }: Props) => {
  const classes = useStyles({});
  useEffect(() => {
    if (!instanceStatus && !isFetching) {
      fetchInstanceStatus();
      isFetching = true;
    }
  });

  return (
    <DashboardNumCard
      displayNum={instanceStatus ? instanceStatus.completedPercentage : null}
      title={I18n.get('cardTitleWorkflowHealth')}
      icon={<SwapVerticalCircle className={classes.cardIcon} />}
      extraContent={(
        <div className={classes.flexDiv}>
          <Box className={classes.box} fontSize={14} color={red[500]}>Failed: {instanceStatus ? instanceStatus.failedPercentage : '--'}</Box>
          <Box fontSize={14} color={lightGreen[500]}>Started: {instanceStatus ? instanceStatus.startedPercentage : '--'}</Box>
        </div>
      )}
    />
  );
};

const mapStateToProps = (state: AppState) => ({ instanceStatus: state.instanceStatus });
const mapDispatchToProps = { fetchInstanceStatus: fetchInstanceStatusAction };
export default connect(mapStateToProps, mapDispatchToProps)(memo(HealthScoreCard));
