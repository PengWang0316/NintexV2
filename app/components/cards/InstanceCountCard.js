import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { orange } from '@material-ui/core/colors';
import { GroupWork } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceCount as fetchInstanceCountAction } from '../../actions/InstanceActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: orange[500],
  },
};

export const InstanceCountCard = ({ classes, instanceCount, fetchInstanceCount }) => {
  useEffect(() => {
    if (!instanceCount) fetchInstanceCount();
  });

  return <DashboardNumCard displayNum={instanceCount} title={I18n.get('cardTitleWorkflowInstances')} icon={<GroupWork className={classes.cardIcon} />} />;
};

InstanceCountCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  instanceCount: PropTypes.number,
  fetchInstanceCount: PropTypes.func.isRequired,
};
InstanceCountCard.defaultProps = {
  instanceCount: null,
};
const mapStateToProps = state => ({ instanceCount: state.instanceCount });
const mapDispatchToProps = { fetchInstanceCount: fetchInstanceCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(InstanceCountCard));
