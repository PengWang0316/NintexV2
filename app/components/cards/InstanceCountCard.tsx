import React, { useEffect, memo, ReactElement } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { orange } from '@material-ui/core/colors';
import { GroupWork } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceCount as fetchInstanceCountAction } from '../../store/InstanceCount/actions';
import { InstanceCount as InstanceCountType } from '../../store/InstanceCount/types';
import { AppState } from '../../store/ConfigureStore';

interface Props {
  instanceCount: InstanceCountType;
  fetchInstanceCount: () => void;
}

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: orange[500],
  },
});
let isFetching = false;

export const InstanceCountCard = (
  { instanceCount = null, fetchInstanceCount }: Props,
): ReactElement => {
  const classes = useStyles({});
  useEffect(() => {
    if (!instanceCount && !isFetching) {
      fetchInstanceCount();
      isFetching = true;
    }
  });

  return <DashboardNumCard displayNum={instanceCount} title={I18n.get('cardTitleWorkflowInstances')} icon={<GroupWork className={classes.cardIcon} />} />;
};

const mapStateToProps = (state: AppState) => ({ instanceCount: state.instanceCount });
const mapDispatchToProps = { fetchInstanceCount: fetchInstanceCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(memo(InstanceCountCard));
