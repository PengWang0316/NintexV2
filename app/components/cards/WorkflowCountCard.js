import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { red } from '@material-ui/core/colors';
import { Stars } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchWorkflowCount as fetchWorkflowCountAction } from '../../actions/WorkflowActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: red[500],
  },
};
let isFetching;

export const WorkflowCountCard = ({ classes, workflowCount, fetchWorkflowCount }) => {
  useEffect(() => {
    if (!workflowCount && !isFetching) {
      fetchWorkflowCount();
      isFetching = true;
    }
  });

  return <DashboardNumCard displayNum={workflowCount} title={I18n.get('cardTitleTotalWorkflows')} icon={<Stars className={classes.cardIcon} />} />;
};

WorkflowCountCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  workflowCount: PropTypes.number,
  fetchWorkflowCount: PropTypes.func.isRequired,
};
WorkflowCountCard.defaultProps = {
  workflowCount: null,
};
const mapStateToProps = state => ({ workflowCount: state.workflowCount });
const mapDispatchToProps = { fetchWorkflowCount: fetchWorkflowCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(WorkflowCountCard));
