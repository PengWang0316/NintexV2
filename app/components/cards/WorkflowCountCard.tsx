import React, { useEffect, memo } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { red } from '@material-ui/core/colors';
import { Stars } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchWorkflowCount as fetchWorkflowCountAction } from '../../store/WorkflowCount/actions';
import { AppState } from '../../store/ConfigureStore';

interface Props {
  workflowCount: number | null;
  fetchWorkflowCount: () => void;
}

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: red[500],
  },
});
let isFetching = false;

export const WorkflowCountCard = ({ workflowCount, fetchWorkflowCount }: Props) => {
  const classes = useStyles({});
  useEffect(() => {
    if (!workflowCount && !isFetching) {
      fetchWorkflowCount();
      isFetching = true;
    }
  });

  return <DashboardNumCard displayNum={workflowCount} title={I18n.get('cardTitleTotalWorkflows')} icon={<Stars className={classes.cardIcon} />} />;
};

const mapStateToProps = (state: AppState) => ({ workflowCount: state.workflowCount });
const mapDispatchToProps = { fetchWorkflowCount: fetchWorkflowCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(memo(WorkflowCountCard));
