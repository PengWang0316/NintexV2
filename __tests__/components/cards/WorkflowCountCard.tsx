import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { WorkflowCountCard } from '../../../app/components/cards/WorkflowCountCard';
import JssProviderWrapper from '../../libs/JssProviderWrapper';

describe('WorkflowCountCard component', () => {
  const defaultProps = {
    workflowCount: null,
    fetchWorkflowCount: jest.fn(),
  };
  beforeEach(() => jest.clearAllMocks());

  test('useEffect with fetchWorkflowCount call', () => {
    render(<WorkflowCountCard {...{ ...defaultProps }} />);
    expect(defaultProps.fetchWorkflowCount).toHaveBeenCalledTimes(1);
  });

  test('Snapshot without workflowCount', () => expect(renderer.create(JssProviderWrapper(<WorkflowCountCard {...{ ...defaultProps }} />)).toJSON()).toMatchSnapshot());
  test('Snapshot with workflowCount', () => expect(renderer.create(JssProviderWrapper(<WorkflowCountCard {...{ ...defaultProps, workflowCount: 10 }} />)).toJSON()).toMatchSnapshot());
});
