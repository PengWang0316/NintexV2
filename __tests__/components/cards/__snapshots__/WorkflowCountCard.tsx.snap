// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowCountCard component Snapshot with workflowCount 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-47 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-contentDiv-49"
    >
      <div>
        <div
          className="MuiBox-root MuiBox-root-91"
        >
          Total Workflows
        </div>
        <div
          className="MuiBox-root MuiBox-root-92"
        >
          10
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-cardIcon-46"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`WorkflowCountCard component Snapshot without workflowCount 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-47 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-contentDiv-49"
    >
      <div>
        <div
          className="MuiBox-root MuiBox-root-80"
        >
          Total Workflows
        </div>
        <div
          className="MuiBox-root MuiBox-root-81"
        >
          --
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-cardIcon-46"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
        />
      </svg>
    </div>
  </div>
</div>
`;
